src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -208,6 +208,121 @@ internal sealed override IEnumerable<Cci.INestedTypeDefinition> GetSynthesizedNe
             return null;
         }
 
+        public sealed override IEnumerable<(Cci.ITypeDefinition, ImmutableArray<Cci.DebugSourceDocument>)> GetTypeToDebugDocumentMap(EmitContext context)
+        {
+            var debugDocuments = ArrayBuilder<Cci.DebugSourceDocument>.GetInstance();
+            var methodDocumentList = PooledHashSet<Cci.DebugSourceDocument>.GetInstance();
+
+            var namespacesAndTypesToProcess = ArrayBuilder<NamespaceOrTypeSymbol>.GetInstance();
```

Comment:
>namespacesAndTypesToProcess [](http://example.com/codeflow?start=16&length=27)

Maybe rename to namespacesAndTopLevelTypesToProcess, which will make it clear that we are not descending into nested types.